docs/contrib/workflow.md

```diff
@@ -0,0 +1,232 @@
+1000+ lines of code.
+
+### Pull request labels
+Your PR must have labels. We have different label categories.
+1. PR type. One of three labels: `feature`, `bugfix`, `improvment`.
+   Pick one which describes your changes best.
+2. Area. `area/testing`, `area/docs`, etc. Describes area which you are changing.
+3. Migration. If your PR contains database migration, you must specify label `DB migration`.
+4. Review status. Label `On review` must be set if your changes are ready for review.
+   If work is still in progress - do not use this label.
+5. Test status. Labels `Ready for Testing` and `tested` were made for QAs to track test
+   status of a PR. These labels will be described in [paragraph](#pull-request-life-cycle)
+   bellow.
+6. Merge status. If your PR has green tests, 2 approvals and label `tested` - it can be marked
+   as `Ready to merge`. It means that this PR can be included into next Release any minute.
+7. Release status. If your RP was merged, a release manager will change `Ready to merge` label 
+   with `Next Release`. You can't set label `Next Release`. Only release menager can.
+
+### Pull request life cycle
+Complete process of PR life cycle looks like:
+
+1. A developer opens a PR with labels from [categories](#pull-request-labels) 1-3.
+2. When the PR is no longer a draft, the developer adds reviewers and puts label `On Review`.
+3. When the task is completed and the PR can be built and deployed, a developer adds 
+   `Ready for Testing` label. It would be a signal for testers to start working on the
+   PR verification. 
+4. After a tester verified the current state of code in the PR (it's not necessary 
+   to have the approved PR at this moment), the tester replaces `Ready for Testing`
+   label with `tested`.
+5. If the tester finds the task is not properly solved (contains bugs, not covers
+   all requirements to close the task or other concerns), the tester removes the
+   `Ready for Testing` label and returns it to the development with the comment
+   which describe what is required to be fixed.
+6. If a developer needs to change something in a tested PR, then after pushing changes, 
+   the developer removes `tested` label and returns `Ready for Testing` label. 
+   This signals to testers that this PR was changed and new testing is needed
+   (the test code might require changes as well). Steps 4-6 can be repeated several times.
+7. When a PR is approved by 2 contributors, the label `tested` was put last time
+   AFTER the pushing of last commit to PR, the developer replaces the `On Review` 
+   label with `Ready to Merge` one.
+8. From this moment a Release manager can merge the PR to include it into release
+   branch. When the PR will be merged the release manager will change `Ready to Merge`
+   label with `Next Release`. Only a release manager can do it.
+
+When you mark your PR as `Ready to Merge` - your job on the PR is done. You can start to work
+on a new one. 
+
+### Pull request merging
```

Comment:
Is there any rules for deleting branches after merging? Any rules for squashing commits? Is it actually rebasing or merging?
At least the deletion policy would be useful. Does setting `ready to merge` also imply `say good bye to your branch`? Can disappear at any moment or will be there forever?

Reply:
Usually I delete a branch after merge. Do we have some reasons not to do it?

Reply:
I don't know. My comment is mostly about expectations: after somebody puts a label `ready to merge` they are not supposed to make any new commits or use that branch in some way. If there is this agreement/expectation, it's worth to write that down.

Reply:
added